adoc/headers/accessorBuffer.h

```diff
@@ -31,6 +31,10 @@ template <typename DataT, int Dimensions = 1,
           access::placeholder isPlaceholder = access::placeholder::false_t>
 class accessor {
  public:
+  static constexpr int dimensions = Dimensions;
+  static constexpr sycl::access_mode access_mode = AccessMode;
```

Comment:
Note that the `sycl::` namespace is not strictly necessary here.  It could be like this:

```
static constexpr access_mode access_mode = AccessMode;
```

But maybe the `sycl::` is better for clarity anyway.

I was concerned at first that this definition of `access_mode` will shadow the type `sycl::access_mode` for any code inside this class.  I think this will not affect the specification, though, because none of the member function parameter or return types use `access_mode`.  Of course, implementations will likely need to change because they undoubtedly use the `access_mode` type in their implementations.  I think that is OK, though.  Raising the shadowing issue here, though, in case others see some problem.

Reply:
I don't know if shadowing is a problem or not, but if it is then I think we already have it, because `multi_ptr` defines:

```c++
static constexpr access::address_space address_space = Space;
```

I'm just pointing that out because if we decide to do something different here, we may need to revisit `multi_ptr`.